packages/react-data-grid/src/Canvas.tsx

```diff
@@ -209,144 +217,60 @@ export default function Canvas<R>({
   }
 
   function getRows() {
-    const rows = [];
-
-    for (let idx = rowOverscanStartIdx; idx <= rowOverscanEndIdx; idx++) {
-      rows.push(renderRow(idx));
-    }
-
-    return rows;
-  }
-
-  function renderRow(idx: number) {
-    const row = rowGetter(idx);
-    const rendererProps: RendererProps<R> = {
-      key: idx,
-      ref(row) {
-        if (row) {
-          rows.set(idx, row);
-        } else {
-          rows.delete(idx);
-        }
-      },
-      idx,
-      row,
-      height: rowHeight,
-      columns: columnMetrics.columns,
-      isRowSelected: isRowSelected(row),
-      onRowSelectionChange,
+    const rowProps = {
       cellMetaData,
-      subRowDetails: getSubRowDetails ? getSubRowDetails(row) : undefined,
-      colOverscanStartIdx,
       colOverscanEndIdx,
-      lastFrozenColumnIndex: columnMetrics.lastFrozenColumnIndex,
+      colOverscanStartIdx,
+      columnMetrics,
+      eventBus,
+      getSubRowDetails,
       isScrolling,
-      scrollLeft
-    };
-    const { __metaData } = row as RowData;
-
-    if (__metaData) {
-      if (__metaData.getRowRenderer) {
-        return __metaData.getRowRenderer(rendererProps, idx);
-      }
-      if (__metaData.isGroup) {
-        return renderGroupRow(rendererProps);
-      }
-    }
-
-    if (rowRenderer) {
-      return renderCustomRowRenderer(rendererProps);
-    }
-
-    return <Row<R> {...rendererProps} />;
-  }
-
-  function isRowSelected(row: R): boolean {
-    return selectedRows !== undefined && selectedRows.has(row[rowKey]);
-  }
+      onRowSelectionChange,
+      rowGetter,
+      rowGroupRenderer,
+      rowHeight,
+      rowKey,
+      rowOffsetHeight,
+      rowRenderer,
+      selectedRows
+    } as const;
+    const rowElements = [];
 
-  function setComponentsScrollLeft(scrollLeft: number) {
-    if (isPositionStickySupported()) return;
-    const { current } = interactionMasks;
-    if (current) {
-      current.setScrollLeft(scrollLeft);
+    for (let idx = rowOverscanStartIdx; idx <= rowOverscanEndIdx; idx++) {
+      rowElements.push(
+        <ActualRow<R>
+          key={idx}
+          idx={idx}
+          {...rowProps}
+        />
+      );
     }
 
-    rows.forEach(row => {
-      if (row && row.setScrollLeft) {
-        row.setScrollLeft(scrollLeft);
-      }
-    });
+    return rowElements;
   }
 
   function getRowTop(rowIdx: number) {
-    const row = rows.get(rowIdx);
-    if (row && row.getRowTop) {
-      return row.getRowTop();
-    }
     return rowHeight * rowIdx;
   }
 
-  function getRowHeight(rowIdx: number) {
-    const row = rows.get(rowIdx);
-    if (row && row.getRowHeight) {
```

Comment:
Maybe this was for dynamic row height consideration? I am not sure if we can remove this.